content/blog/2022/2022-07-03-getting-started-with-aws-step-functions.md

```diff
@@ -0,0 +1,549 @@
+|`process payment`|payment type, cart items|Price of each item|Fetching Price of cart items|
+|`create order`|cstomer ID, cart items with price|`success/failure`|Create order if payment is successful|
+
+When we design the order in which to execute these steps, we need to consider which steps can execute in parallel and which ones are in sequence. 
+
+We can execute the steps for `fetch customer` and `fetch price` in parallel. Once these two steps are complete, we will execute the step for `process payment`. If the payment fails, we will end the checkout process with an error. If the payment succeeds, we will create an order for the customer. The step for `process payment` can be retried a specific number of times on failures.
+
+We will use AWS Step Function to represent this orchestration in the next sections.
+
+## Creating the Lambda Functions for Invoking from the State Machine
+Let us define skeletal Lambda functions for each of the steps of the `checkout` process. Our Lambda function to fetch the customer data looks like the following:
+
+```js
+exports.handler = async (event, context, callback) => {
+    console,log(`input: ${event.customer_id}`)
+    // TODO fetch from database
+    callback(null,
+                {
+                    customer_id: event.customer_id, 
+                    customer_name: "John Doe", 
+                    payment_pref: "Credit Card",
+                    email: "john.doe@yahoo.com", 
+                    mobile: "677896678"
+                }
+            )
+};
+
+```
+This lambda takes `customer_id` as input and returns a customer record corresponding to the `customer_id`. Since the lambda function is not the focus of this post, we are returning a hardcoded value of customer data instead of fetching it from the database.
+
+Similarly, our Lambda function for fetching price looks like this:
+
+```js
+exports.handler = async (event, context, callback) => {
+    const item_no = event.item_no
+    console.log(`item::: ${item_no}`)
+   
+   // TODO fetch price from database
+    const price = {item_no: item_no, price: "123.45", lastUpdated: "2022-06-12"}
+    callback(null, price)
+}
+
+```
+This Lambda takes `item_no` as input and returns a pricing record corresponding to the `item_no`. 
+
+We will use similar Lambda functions for the other steps of the `checkout` process. All of them will have skeletal code similar to the `fetch customer` and `fetch price`.
+
+## Defining the Checkout Process with a State Machine
+After defining the Lambda functions and getting an understanding of the basic concepts of the Step Function service, let us now define our `checkout` Process. 
+
+Let us create the state machine from the AWS management console. We can either choose to use the visual workflow editor or ASL for defining our state machine. 
+{{% image alt="checkout process" src="images/posts/aws-step-function/create_state_machine.png" %}}
```

Comment:
Pleas make single screenshots for the steps. It is hard to understand, when all stepas are on one picture.